eth/fetcher/fetcher.go

```diff
@@ -338,9 +338,15 @@ func (f *Fetcher) loop() {
 			if _, ok := f.fetching[notification.hash]; ok {
 				break
 			}
+			if _, ok := f.fetched[notification.hash]; ok {
+				break
+			}
```

Comment:
I don't see why this change is useful.

In the fetcher, originally new announcements are accumulated into `announces`.

When enough time passed, a single announcement in converted into the header fetch request, and it's placed into `fetching`, with the result ending up in `fetched`. Note, `fetching` does not get cleared when the result arrives and is placed into `fetched`.

Eventually, items are moved from `fetched` to `completing` and only at this point are they evacuated from `fetching`. As such, when `fetching` becomes "empty", the item is already inside `completing`.